### PR TITLE
Add fields for filtering cloudbackup enumerate.

### PR DESCRIPTION
**Pull request by talakad**
Also add support for pagination of enumerate output.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -343,6 +343,10 @@ type CloudBackupGenericRequest struct {
 	CredentialUUID string
 	// All if set to true, backups for all clusters in the cloud are processed
 	All bool
+	// StatusFilter indicates backups based on status
+	StatusFilter CloudBackupStatusType
+	// TagFilter indicates backups based on tag
```
Comment (disrani-px) [inline api/api.go]: Not sure if we need tag here, it is an implementation detail in portworx that isn't exposed in openstorage

Comment (talakad) [reply api/api.go]: Perhaps i should rename this to metadataType, to allow implementations to have their own filter based on metadata?

Comment (talakad) [reply api/api.go]: Changed this to metadata filters.

File: api/server/sdk/cloud_backup.go [hunk 2]
```diff
@@ -220,14 +220,22 @@ func (s *CloudBackupServer) EnumerateWithFilters(
...
 			CredentialUUID: credId,
 			All:            req.GetAll(),
+			MetadataFilter: req.MetadataFilter,
 		},
-	})
+		ContinuationToken: req.ContinuationToken,
+		MaxBackups:        req.MaxBackups,
+	}
+	if req.StatusFilter != api.SdkCloudBackupStatusType_SdkCloudBackupStatusTypeUnknown {
+		enumerateReq.StatusFilter = api.CloudBackupStatusType(api.SdkCloudBackupStatusTypeToCloudBackupStatusString(req.StatusFilter))
```
Comment (disrani-px) [inline api/server/sdk/cloud_backup.go]: Can we use a bigger API name for `SdkCloudBackupStatusTypeToCloudBackupStatusString`?

**Pull request merged**